sphinx_search/_static/js/rtd_sphinx_search.js

```diff
@@ -406,19 +418,74 @@ const removeSearchModal = () => {
     search_outer_wrapper.classList.remove("display-block");
 };
 
+/**
+ * Returns the search URL
+ *
+ * @param {Object} searchParams params for search
+ */
+const getSearchUrl = searchParams => {
+    const api_host = getProjectInfo().api_host;
+    const url =
+        api_host + "/api/v2/docsearch/?" + convertObjToUrlParams(searchParams);
+    return url;
+};
+
+/**
+ * Returns project info
+ */
+const getProjectInfo = () => {
+    return {
+        project: READTHEDOCS_DATA.project,
+        version: READTHEDOCS_DATA.version,
+        language: READTHEDOCS_DATA.language || "en",
+        api_host: READTHEDOCS_DATA.api_host
+    };
+};
+
+/**
+ * Returns the URL parameters in the form of Object
+ *
+ * @return {Object} url params
+ */
+const getUrlVars = () => {
+    let vars = {};
+    let params = window.location.href.replace(
+        /[?&]+([^=&]+)=([^&]*)/gi,
+        (m, key, value) => {
+            vars[key] = value;
+        }
+    );
+    return vars;
```

Comment:
Is this really not in jQuery already? I feel like we're reinventing a lot of wheels.

Reply:
No.
I checked on the internet. There is no direct utility from the jquery or javascript to get the url parameters.